### PR TITLE
HDDS-3777. Reduce recon UI build time

### PR DESCRIPTION
**Pull request by vivekratnavel**
## What changes were proposed in this pull request?

- Replace yarn as the default package manager to pnpm
- Replace -Dskip.yarn flag with -Dskip.npx
- Use caching in Github Actions workflow to cache node_modules and pnpm-store.
- Start running acceptance tests only after build succeeds

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3777

## How was this patch tested?

This patch was tested with several CI triggers. It reduces the recon build time significantly from ~16 minutes to less than 2 minutes. 

Example run - https://github.com/vivekratnavel/hadoop-ozone/runs/763625356?check_suite_focus=true

### CODE REVIEW

File: hadoop-ozone/recon/pom.xml [hunk 1]
```diff
@@ -86,37 +86,46 @@
       <plugin>
         <groupId>com.github.eirslett</groupId>
         <artifactId>frontend-maven-plugin</artifactId>
+        <version>1.10.0</version>
```
[the remark this answers is not in the document]
Comment (vivekratnavel) [reply hadoop-ozone/recon/pom.xml:89]: Good catch! I Will fix it.

File: hadoop-ozone/recon/pom.xml [hunk 2]
```diff
@@ -86,7 +86,7 @@
       <plugin>
         <groupId>com.github.eirslett</groupId>
         <artifactId>frontend-maven-plugin</artifactId>
-        <version>1.10.0</version>
+        <version>${frontend-maven-plugin.version}</version>
```
Comment (adoroszlai) [inline hadoop-ozone/recon/pom.xml:89]: Nit: version is inherited from parent pom's `pluginManagement` section, no need to explicitly specify here.  But it does no harm and can be removed later, it's more important to merge this speed improvement sooner than later.

**Comment by adoroszlai** [comment]
@elek can we merge this?

**Comment by vivekratnavel** [comment]
@adoroszlai Thanks for the review! I have merged it now. If there is any issue, we can always revert it.

**Comment by elek** [comment]
Thanks for the improvement.

(Note: As I remember the License generator for frontend was a built-in yarn feature. We might need to find the best npm based approach.)

**Pull request merged**